src/main/java/com/denizenscript/ddiscordbot/commands/DiscordCreateChannelCommand.java

```diff
@@ -0,0 +1,103 @@
+        setRequiredArguments(3, 6);
+    }
+    // <--[command]
+    // @Name discordcreatechannel
+    // @Syntax discordcreatechannel [id:<id>] [group:<group_id>] [name:<name>] (description:<description>) (category:<category_id>) (position:<#>)
+    // @Required 3
+    // @Maximum 6
+    // @Short Creates text channels on Discord.
+    // @Plugin dDiscordBot
+    // @Group external
+    //
+    // @Description
+    // Creates text channels on Discord.
+    //
+    // You can specify the channel "topic" (description) with the "description" argument.
+    //
+    // You can specify the channel's parent category with the "category" argument.
+    // By default, the channel will not be attached to any category.
+    //
+    // You can specify the channel's position in the list as an integer with the "position" argument.
+    //
+    // The command should usually be ~waited for. See <@link language ~waitable>.
+    //
+    // @Tags
+    // <entry[saveName].channel> returns the DiscordChannelTag of a channel upon creation when the command is ~waited for.
+    //
+    // @Usage
+    // Use to create a channel in a category.
+    // - ~discordcreatechannel id:mybot group:1234 name:my-channel category:5678
+    //
+    // @Usage
+    // Use to create a channel and log its name upon creation.
+    // - ~discordcreatechannel id:mybot group:1234 name:very-important-channel save:stuff
+    // - debug log "Created channel '<entry[stuff].channel.name>'"
+    //
+    // -->
+
+    @Override
+    public void parseArgs(ScriptEntry scriptEntry) {
+    }
+
+    @Override
+    public void execute(ScriptEntry scriptEntry) {
+        ElementTag id = scriptEntry.requiredArgForPrefixAsElement("id");
+        DiscordGroupTag group = scriptEntry.requiredArgForPrefix("group", DiscordGroupTag.class);
+        ElementTag name = scriptEntry.requiredArgForPrefixAsElement("name");
+        ElementTag description = scriptEntry.argForPrefixAsElement("description", null);
+        ElementTag category = scriptEntry.argForPrefixAsElement("category", null);
+        ElementTag position = scriptEntry.argForPrefixAsElement("position", null);
+        if (scriptEntry.dbCallShouldDebug()) {
+            Debug.report(scriptEntry, getName(), id, group, name, description, category);
+        }
+        if (group != null && group.bot == null) {
+            group.bot = id.asString();
+        }
+        Bukkit.getScheduler().runTaskAsynchronously(DenizenDiscordBot.instance, () -> {
+            try {
+                ChannelAction<TextChannel> action = group.getGuild().createTextChannel(name.asString());
+                if (description != null) {
+                    action = action.setTopic(description.asString());
+                }
+                if (category != null) {
+                    Category resultCategory = group.getGuild().getCategoryById(category.asString());
+                    if (resultCategory == null) {
+                        Debug.echoError("Invalid category!");
```

Comment:
has to be scheduled onto main thread, and also include the `scriptEntry` reference as the first param

Reply:
How do I accomplish the first, and what does the second mean?

Reply:
look at `DiscordCreateThreadCommand`

Reply:
I've pushed a working change, but I still don't know what the latter part of your initial comment means